Issue-384 added reference to core module in pom.xml

Casva opened the pull request:
#384 Added reference to andhow core to [this pom file](https://github.com/eeverman/andhow/blob/master/andhow-testing/andhow-annotation-processor-test-harness/pom.xml)

eeverman commented:
Thanks for the contribution, @Casva 
There actually a bit more to this task:
* The class `org.yarnandtail.compile.IOUtil` actually needs to be removed/deleted.
* The one place where the deleted class is used needs to be updated to use the other IOUtil class.

Casva commented:
Ah my mistake @eeverman I misread.

- I've deleted [the old IOUtil file](https://github.com/eeverman/andhow/blob/master/andhow-testing/andhow-annotation-processor-test-harness/src/main/java/org/yarnandtail/compile/IOUtil.java) and...
- Added a reference to [the kept IOUtil](https://github.com/eeverman/andhow/blob/master/andhow-core/src/main/java/org/yarnandtail/andhow/util/IOUtil.java) on the [one test](https://github.com/eeverman/andhow/blob/master/andhow-testing/andhow-annotation-processor-test-harness/src/main/java/org/yarnandtail/compile/TestSource.java) that I found that used it.

Hope everything is good now! 😄
